src/IO/WriteHelpers.h

```diff
@@ -581,6 +581,59 @@ void writeCSVString(const StringRef & s, WriteBuffer & buf)
     writeCSVString<quote>(s.data, s.data + s.size, buf);
 }
 
+inline void writeHTMLString(const char * begin, const char * end, WriteBuffer & buf)
+{
+    const char * pos = begin;
+    while (true)
+    {
+        const char * next_pos = find_first_symbols<'<', '&', '>', '"', '\''>(pos, end);
```

Comment:
What about control characters? Should we replace \0 to `&#0;` or it doesn't make sense?

Reply:
In XML there are only five predefined character entity references, so I think it is not necessary.